pkg/controller/template/template_controller.go

```diff
@@ -171,12 +182,54 @@ func (ctrl *Controller) deleteSecret(obj interface{}) {
 	}
 }
 
+func (ctrl *Controller) enqueueController() {
+	cfg, err := ctrl.ccLister.Get(ctrlcommon.ControllerConfigName)
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("couldn't get ControllerConfig on secret callback %#v", err))
+		return
+	}
+	glog.V(4).Infof("Re-syncing ControllerConfig %s due to secret change", cfg.Name)
+	ctrl.enqueueControllerConfig(cfg)
+}
+
+func (ctrl *Controller) updateFeature(old, cur interface{}) {
+	oldFeature := old.(*osev1.FeatureGate)
+	newFeature := cur.(*osev1.FeatureGate)
+	if !reflect.DeepEqual(oldFeature.Spec, newFeature.Spec) {
+		glog.V(4).Infof("Update Feature %s", newFeature.Name)
+		ctrl.enqueueController()
+	}
+}
+
+func (ctrl *Controller) addFeature(obj interface{}) {
+	features := obj.(*osev1.FeatureGate)
+	glog.V(4).Infof("Adding Feature %s", features.Name)
+	ctrl.enqueueController()
+}
+
+func (ctrl *Controller) deleteFeature(obj interface{}) {
+	features, ok := obj.(*osev1.FeatureGate)
+	if !ok {
+		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			utilruntime.HandleError(fmt.Errorf("Couldn't get object from tombstone %#v", obj))
+			return
+		}
+		features, ok = tombstone.Obj.(*osev1.FeatureGate)
+		if !ok {
+			utilruntime.HandleError(fmt.Errorf("Tombstone contained object that is not a FeatureGate %#v", obj))
+			return
+		}
+	}
+	glog.V(4).Infof("Deleted Feature %s and restored default controller config", features.Name)
+}
```

Comment:
Do we not need to enqueue a reconcile if the feature is deleted?

Reply:
No, it is usual for such static resources to be re-queued only on add/update. Removing it and reconciling is as if you rip off a part of cluster, and therefor destroying it's health.

Reply:
This affects the state of the output of MCO though, so deleting it does infer a different desired state, is something going to automatically put it back if you delete it?  I'll leave this decision to the MCO owners but to me it seems natural to reconcile this on delete as well